src/components/ResultTypes.js

```diff
@@ -0,0 +1,33 @@
+export class CourseItemResult {
+    /**
+     * Creates a CourseItemResult from the given search result data
+     * @param {String} id
+     * @param {String} description
+     * @param {String} pageName
+     * @param {String} siteName
+     */
+    constructor(id, description, pageName, siteName) {
+        this.url = id;
+        this.description = description;
+        this.pageName = name;
```

Comment:
This should be `this.pageName = pageName` right

Reply:
Oh damn... I wonder why that didn't produce an error. I guess it just does `this.pageName = undefined`.